onnx_caffe2/backend.py

```diff
@@ -305,6 +306,68 @@ def _create_optimized_rnn(cls, n):
             else 'linear')
         return op
 
+    @classmethod
+    def _create_slice(cls, n):
+        op = cls._common_onnx_node_to_caffe2_op(n)
+        data, axes, orig_starts, orig_ends = op.input
+        ops = []
+
+        data_shape = dummy_name()
+        ops.append(core.CreateOperator(
+            'Shape',
+            [data],
+            [data_shape]
+        ))
+
+        tmp_starts = dummy_name()
+        starts = dummy_name()
+        ops.extend([
+            core.CreateOperator(
+                'ConstantFill',
+                [data_shape],
+                [tmp_starts],
+                value=0,
+            ),
+            core.CreateOperator(
+                'ScatterAssign',
+                [tmp_starts, axes, orig_starts],
```

Comment:
This is a little strange here. Can we do the cast first, then do ScatterAssign?

Reply:
Why is it strange? I should mention that doing Cast is because Slice only supports INT32.

Reply:
I see why the cast is needed. Do we need to declare the dtype of ConstantFill op?

Reply:
ConstantFill can infer the type from value. But I like the idea being more explicit :-) will add dtype=INT64